test/cpp/util/test_credentials_provider.cc

```diff
@@ -115,7 +121,22 @@ class DefaultCredentialsProvider : public CredentialsProvider {
     } else if (type == grpc::testing::kAltsCredentialsType) {
       grpc::experimental::AltsServerCredentialsOptions alts_opts;
       return grpc::experimental::AltsServerCredentials(alts_opts);
-    } else if (type == grpc::testing::kTlsCredentialsType) {
+    } else if (type == grpc::testing::kTls12CredentialsType) {
```

Comment:
Could we combine the `else if`'s for TLS 1.2 and 1.3 (to avoid a lot of duplicate code) and then have an additional if statement when we go to set the min/max TLS version?